Dockerfile

```diff
@@ -1,74 +1,137 @@
+    && chmod 644 $PROJECTOR_DIR/ide/projector-server/lib/*
 
-# Copy predefined configs
-COPY --chown=0:0 config/etc /etc/
+# Stage 4. Build the main image with necessary environment for running Projector
+#   Doesn't require to be a desktop environment. Projector runs in headless mode.
+FROM fedora:32
+ARG downloadUrl
+ENV PROJECTOR_USER_NAME projector-user
+ENV PROJECTOR_DIR /projector
+ENV HOME /home/$PROJECTOR_USER_NAME
+ENV PROJECTOR_CONFIG_DIR $HOME/.config
+RUN true \
+    # Any command which returns non-zero exit code will cause this shell script to exit immediately.
+    && set -e \
+    # Activate debugging to show execution details: all commands will be printed before execution.
+    && set -x \
+    # Install necessary packages.
+    && yum install -y \
+    # The non-interactive network downloader.
+    wget \
+    # Git is a fast, scalable, distributed revision control system with an unusually rich command set that provides both high-level operations and full access to internals.
+    git \
+    # Network Security Services (NSS) is a set of libraries designed to support cross-platform development of security-enabled client and server applications. 
+    # Applications built with NSS can support SSL v3, TLS, PKCS #5, PKCS #7, PKCS #11, PKCS #12, S/MIME, X.509 v3 certificates, and other security standards.
+    nss \
+    # Packages required by JetBrains products.
+    libsecret jq e2fsprogs-libs \
+    # Packages needed for AWT.
+    libXext libXrender libXtst libXi freetype \
+    # Collection of shell functions that take advantage of the programmable completion feature of bash 2.04 and later.
+    bash-completion \
+    # Provides command line and full screen utilities for browsing procfs, a "pseudo" file system dynamically generated by the kernel to provide information about the status 
+    # of entries in its process table (such as whether the process is running, stopped, or a "zombie"). Requires by IDEA to disable warnings. 
+    procps \
+    # Provides find utility. Fedora 32 doesn't contain find utility by default.
+    findutils \
+    # Check whether download Url represent IntelliJ IDEA to install required dependencies.
+    && if [ "${downloadUrl#*idea}" != "$downloadUrl" ]; then yum install -y java-11-openjdk-devel; else echo "Not IntelliJ Idea"; fi \
+    # Check whether download Url represent PyCharm to install required dependencies.
+    && if [ "${downloadUrl#*pycharm}" != "$downloadUrl" ]; then yum install -y python2 python3 python3-distutils-extra python3-pip python3-setuptools; else echo "Not Pycharm"; fi
 
-# Copy sh scripts
-COPY --chown=0:0 scripts/*.sh /opt/
+RUN true \
+    # Any command which returns non-zero exit code will cause this shell script to exit immediately.
+    && set -e \
+    # Activate debugging to show execution details: all commands will be printed before execution.
+    && set -x \
+    && adduser -r -u 1001 -G root -d $HOME -m -s /bin/sh $PROJECTOR_USER_NAME \
```

Comment:
does this work on Openshift? could we use the same anyuid stuff we have for https://github.com/eclipse-che/che-devfile-registry/blob/master/arbitrary-users-patch/entrypoint.sh ?

Reply:
It has been done in the same manner as in theia's image: https://github.com/eclipse-che/che-theia/blob/master/dockerfiles/theia/docker/ubi8/runtime-adduser.sh
It was checked on Dogfooding instance.